com.ibm.streamsx.inet/com.ibm.streamsx.inet.http/HTTPParallelRequest.spl

```diff
@@ -0,0 +1,134 @@
+		 */
+		static HttpMethod = enum {httpGet, httpPut, httpPost, httpDelete};
+
+	graph
+		@parallel(width=$width)
+		(
+			stream<I, tuple<rstring body, list<rstring> headers, rstring message, int32 status, rstring url>> Result as R;
+			stream<rstring url, rstring message, int32 status> Error as E
+		) = Custom(Request as I) {
+			logic
+				onTuple I: {
+					mutable int32 error_ =0;
+					mutable list<rstring> replyHeaders_ = [];
+					mutable list<rstring> requestHeaders_ = $headers;
+					if (length($contentTypeHeader) > 0) {
+						appendM(requestHeaders_, $contentTypeHeader);
+					}
+					rstring urlProcessed_ = $urlEncode($url);
+					mutable rstring result_ = "";
+					if (isTraceable(Trace.trace)) {
+						appTrc(Trace.trace, "Emit http request "+
+									(rstring)$method+
+									" url:"+urlProcessed_+
+									" requestHeaders:"+(rstring)requestHeaders_+
+									" user:"+$user+
+									" password:*****"+
+									" certFile:"+$certFile+
+									" certType:"+$certType+
+									" keyFile:"+$keyFile+
+									" keyType:"+$keyType+
+									" keyPass:"+$keyPass+
+									" requestTimeout:"+(rstring)$requestTimeout+
+									" connectionTimeout:"+(rstring)$connectionTimeout);
+					}
+					if ($method == HttpMethod.httpGet) {
+						result_ = com.ibm.streamsx.inet.http::httpGet   (       urlProcessed_, requestHeaders_, $user, $password, $certFile, $certType, $keyFile, $keyType, $keyPass, replyHeaders_, error_, $requestTimeout, $connectionTimeout);
+					} else if ($method == HttpMethod.httpPost) {
+						result_ = com.ibm.streamsx.inet.http::httpPost  ($data, urlProcessed_, requestHeaders_, $user, $password, $certFile, $certType, $keyFile, $keyType, $keyPass, replyHeaders_, error_, $requestTimeout, $connectionTimeout);
+					} else if ($method == HttpMethod.httpPut) {
+						result_ = com.ibm.streamsx.inet.http::httpPut   ($data, urlProcessed_, requestHeaders_, $user, $password, $certFile, $certType, $keyFile, $keyType, $keyPass, replyHeaders_, error_, $requestTimeout, $connectionTimeout);
+					} else if ($method == HttpMethod.httpDelete) {
+						result_ = com.ibm.streamsx.inet.http::httpDelete(       urlProcessed_, requestHeaders_, $user, $password, $certFile, $certType, $keyFile, $keyType, $keyPass,                error_, $requestTimeout, $connectionTimeout);
+					}
+					mutable rstring mess_="success";
+					if (error_ != 0) {
+						mess_="error";
+						appTrc(Trace.error,"Error code in method "+(rstring)$method+" url:"+urlProcessed_+" was "+(rstring)error_);
+						submit({url=urlProcessed_, message=mess_, status=error_}, E);
+					}
+					mutable tuple<R> ot = {};
+					assignFrom(ot, I);
+					ot.body=result_; ot.headers=replyHeaders_; ot.message=mess_; ot.status=error_;
+					submit(ot, R);
+				}
+				onPunct I : {
+					submit(currentPunct(), R);
+					submit(currentPunct(), E);
+				}
+			config
+				threadedPort : queue(I, $congestionPolicy, $queue);
+		}
+}
+
+/**
```

Comment:
Why would `passThrough` ever be needed (or a good idea)?

Just seems to me that URLs must be URLencoded, so why not force that to happen and avoid the potential of an error due to `passThrough`, especially one that occurs when the data becomes changes in a running app. E.g. original testing only used ASCII but changes in the data source leads to non-ASCII or problematic characters.

Reply:
In almost all cases the function urlEncode is used, I think. This is the default.
The passThrough is useful in cases where the url handling id done from an upstream function. 
Probably the user wants to implement an url-check (like the InetSource operator). But in issue #297 I learned that in some cases the check should be relaxed. I introduced a variable (user defined) to enable the full flexibility here.
The comment should name the default function in the first place.